### PR TITLE
Increase flow coverage and refactor controller.

### PR DESCRIPTION
**Pull request by arne-z**
See title.

### CODE REVIEW

File: src/controller/CarouselFactory.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+// @flow
+import { Image as DialogflowImage, BrowseCarousel, BrowseCarouselItem } from 'actions-on-google';
+import logger from '../logger';
+import type { convParams } from './DialogflowController';
+import type{ Response, Image } from './RequestController';
```
Comment (erksch) [inline src/controller/CarouselFactory.js]: Missing space between braces and `type` here:  `import type { ... }`

File: src/controller/CarouselFactory.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+export function getGoodImages(response: Response) {
+  const { images } = response.data;
+  logger.info(`Images are: ${JSON.stringify(images)}`);
+  const goodImages: Array<Image> = images.filter(element => element.score > THRESHHOLD);
+  logger.info(`good Images are: ${JSON.stringify(goodImages)}`);
+  return goodImages;
+}
+
+export function getHumanReadableTags(image: Image): string {
```
Comment (erksch) [inline src/controller/CarouselFactory.js]: Do you need to export all these functions?

Comment (erksch) [reply src/controller/CarouselFactory.js]: If you do maybe this is not the right file for them.

File: src/controller/RequestController.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import logger from '../logger';
+import type{ convParams } from './DialogflowController';
+
+const url: string = 'https://api.askir.me/images';
+
+export type Image = {
+  id: string,
+  type: string,
+  created_at: string,
+  updated_at: string,
```
Comment (erksch) [inline src/controller/RequestController.js]: Personally, I am a fan of converting snake case params from other APIs to javascript camel case.

File: src/controller/RequestController.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  if (params.Tag) { requestString.params.search = params.Tag; }
+  if (params.DatePeriod) { requestString.params.start_date = params.DatePeriod.startDate; }
+  if (params.DatePeriod) { requestString.params.end_date = params.DatePeriod.endDate; }
+  if (params.Author) { requestString.params.author = params.Author; }
+  logger.info(JSON.stringify(requestString));
+  return requestString;
+}
+
+export default function request(params: convParams): Promise<Response> {
+  return axios.get(url, setParams(params));
```
Comment (erksch) [inline src/controller/RequestController.js]:
```
const response = await axios.get(url, setParams(params));
return humps.camelizeKeys(response.data);
```

File: src/controller/RequestController.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import type{ convParams } from './DialogflowController';
+
+const url: string = 'https://api.askir.me/images';
+
+export type Image = {
+  id: string,
+  type: string,
+  created_at: string,
+  updated_at: string,
+  tags: Array<string>, file_date: string,
```
Comment (erksch) [inline src/controller/RequestController.js]: Newline pls

File: src/controller/DialogflowController.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
+// @flow
+import { Conversation } from 'actions-on-google';
+import logger from '../logger';
+import request from './RequestController';
+import { makeCarousel, makeImage } from './CarouselFactory';
+import type { ResponseData, Image, ConvParams } from '../types';
+
+
+const THRESHHOLD: number = 0;
```
Comment (erksch) [inline src/controller/DialogflowController.js]: *THRESHOLD

Comment (arne-z) [reply src/controller/DialogflowController.js]: ooops

**Pull request merged**